### PR TITLE
done

### PR DESCRIPTION
**Pull request by shawnmne**

### CODE REVIEW

File: lib/strings.md [hunk 1]
```diff
@@ -5,20 +5,33 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
...
 
-Replace me with your response...
+Strings are a sequence of characters denoted by single or double quotes.
 
 # What are some examples of information that would be Strings as opposed to some other data type?
 
-Replace me with your response...
+name = "Shawn Michaud"
+address = "4245 South St."
+wifename = "married to Melissa"
```
Comment (andrewek) [inline lib/strings.md]: > `wifename = "married to Melissa"` 

Stylistic note: if you've got a variable whose name is more than one word smashed together, stick an underscore in there. This should be:

`wife_name = "Melissa"`

No need to fix it here, but this is good to know going forward.

File: lib/strings.md [hunk 2]
```diff
@@ -5,20 +5,33 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
...
+puts name[5]
+puts name[7]
+
+When trying to retrieve a character that doesn't exist it will return nil
+
 
 # The previous question asks about finding, for example, the 6th character in a String. Is it possible to find the **-6th** (Notice the negative symbol!) character in a String? What does that even mean?
 
-Replace me with your response...
+It is not possible.  Strings start at 0.  It will return nil as above. However you could index back from a point in a string but I'm not sure why you would want to do this.
```
Comment (andrewek) [inline lib/strings.md]: > It is not possible....

Just for kicks, try out something like `name[ -6 ]` and see what happens.

File: lib/strings.md [hunk 3]
```diff
@@ -5,20 +5,33 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
...
+name = "Ada Lovelace"
+puts name[5,-3]
+output would be the same as 
+puts name[2]
 
 # What is one way, using Ruby, to replace certain characters in a string with some other set of characters? For example, given `"Sumeet Jain"`, how would you replace all of the `e` characters in my name with exclamation marks? (So it would be `"Sum!!t Jain"`.)
 
-Replace me with your response...
+The string operator .sub will replace a first occurence of a string to be replaced .gsub will do it globally.
+puts "Sumeet Jain".sub("ee", "??") would work but also
```
Comment (andrewek) [inline lib/strings.md]: You've got `?` where you probably need `!`

**Comment by andrewek** [comment]
@shawnmne Good stuff so far. A few things to look at and/or revise. Make the changes, commit, and re-push to GitHub. The Pull request will update automatically.

**Comment by shawnmne** [comment]
Andrew,
I have updated and saved the file on cloud 9.  I tried using git commit and git push in the command lines but I'm not sure that they are the correct commands.  I also tried git commit -m "update".What should I be using?
Thanks,
Shawn

Date: Tue, 23 Feb 2016 14:52:21 -0800

**Comment by andrewek** [comment]
Shawn.

Try this sequence:

1. `git add -A`
2. `git commit -m "whatever your message is"
3. `git push origin master`

Before and after Step 1, you can type `git status` to see things -- the first time, you should see your file as "Unstaged" (or something like that). The second time you should see it as staged. If you run the command again after you commit, you should see no changes pending or staged.

**Comment by andrewek** [comment]
Awesome. You're done with this one.

**Pull request closed without merging**